rails/app/models/theme.rb

```diff
@@ -9,9 +9,19 @@ class Theme < ApplicationRecord
   validates :sponsor_logos, blob: { content_type: ['image/png', 'image/jpg', 'image/jpeg'], size_range: 1..5.megabytes }
   validates :mapbox_access_token, :presence =>  {:message => 'is required when the Mapbox style URL is set.'}, unless: -> { mapbox_style_url.blank? }
   validates :mapbox_style_url, :presence => {:message => 'is required when the Mapbox access token is set.'}, unless: -> { mapbox_access_token.blank? }
-  validates :center_lat, :sw_boundary_lat, :ne_boundary_lat,
+  validate :map_bounds
+   # validate bounding box if all four values are nil OR if all four values are numeric & in the proper range
+   def map_bounds
+     return if sw_boundary_long.nil? && sw_boundary_lat.nil? && ne_boundary_long.nil? && ne_boundary_lat.nil?
+
```

Comment:
I would add one more validation here:

`errors.add(:base, "All four bounding box values must be set, or left blank")`

Reply:
I just committed a change to add this validation! The message will show if the user submits a mixture of nil values and numeric values.